storage/connect/reldef.cpp

```diff
@@ -54,6 +58,44 @@
 extern handlerton *connect_hton;
 #endif   // !__WIN__
 
+/* aix dladdr equiv */
+
+#if defined (_AIX)
+struct Dl_info {
+  const char* dli_fname;
+};
+int dladdr(void* s, Dl_info* i) {
+   static const size_t bufSize = 4096;
+   void* buf = malloc(bufSize);
+   char* pldi = (char*)buf;
+   int r = loadquery(L_GETINFO,  pldi,  bufSize);
+   if (r == -1) {
+      i->dli_fname = 0;
+      free(buf);
+      return 0;
+   }
+   // First is main(), skip.
+   ld_info* ldi = (ld_info*)pldi;
+   while (ldi->ldinfo_next) {
+     pldi += ldi->ldinfo_next;
+     ldi = (ld_info*)pldi;
+     char* textBegin = (char*)ldi->ldinfo_textorg;
+     if (textBegin < s) {
+        char* textEnd = textBegin + ldi->ldinfo_textsize;
+        if (textEnd > s) {
+           i->dli_fname = ldi->ldinfo_filename;
+           free(buf);
+           return 1;
+        }
+     }
+   }
+   i->dli_fname = 0;
+   free(buf);
+   return 0;
+}
+#endif
+
+
```

Comment:
duh...built with the actual unmodified file from git, and a completely new error comes up. 

`[ 65%] Building CXX object storage/connect/CMakeFiles/connect.dir/reldef.cpp.o
cd /usr/local/src/mariadb-10.1-aix-github/mariadb-aix-build/storage/connect && /opt/IBM/xlC/13.1.0/bin/xlC_r  -+ -DFORCE_INIT_OF_VARS -DHAVE_CONFIG_H -DHUGE_SUPPORT -DLIBXML2_SUPPORT -DLINUX -DMARIADB -DMYSQL_DYNAMIC_PLUGIN -DPIVOT_SUPPORT -DUBUNTU -DUNIX -DXMAP -DZIP_SUPPORT -Dconnect_EXPORTS -q64 -qlanglvl=extended0x -qmaxmem=-1 -qstaticinline -qcpluscmt -qarch=pwr7 -qtune=pwr7 -Dalloca=__alloca -D_H_ALLOCA -DNDEBUG -DSYSV -D_AIX -D_AIX64 -D_AIX61 -D_AIX71 -D_ALL_SOURCE -DUNIX -DFUNCPROTO=15 -O2 -DWITH_INNODB_DISALLOW_WRITES  -g -D_FORTIFY_SOURCE=2 -DDBUG_OFF -qpic -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix-build/include -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/include -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/sql -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix-build/pcre -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/pcre -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/extra/yassl/include -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/extra/yassl/taocrypt/include -I/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/zlib -I/usr/local/itsvbuild/64/include/libxml2    -o CMakeFiles/connect.dir/reldef.cpp.o -c /usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/xindex.h", line 69.12: 1540-1608 (W) An anonymous union should only define non-static data members.
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 87.6: 1540-2963 (S) The member "bool Define(PGLOBAL, PCATLG, LPCSTR, LPCSTR)" does not match any member declared in its containing class.
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.h", line 101.22: 1540-2964 (I) Candidate member in the containing class: "bool Define(PGLOBAL, PCATLG, LPCSTR, LPCSTR, LPCSTR)".
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 191.21: 1540-0217 (S) "GetSizeCatInfo" is not a member of "class CATALOG".
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 196.20: 1540-0215 (S) The wrong number of arguments has been specified for "TABDEF::Define(PGLOBAL, PCATLG, LPCSTR, LPCSTR, LPCSTR)".
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 221.15: 1540-0217 (S) "GetStringCatInfo" is not a member of "class CATALOG".
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 222.16: 1540-0217 (S) "GetStringCatInfo" is not a member of "class CATALOG".
"/usr/local/src/mariadb-10.1-aix-github/mariadb-aix/storage/connect/reldef.cpp", line 272.32: 1540-0217 (S) "UseTemp" is not a member of "struct <unnamed>".
storage/connect/CMakeFiles/connect.dir/build.make:662: recipe for target 'storage/connect/CMakeFiles/connect.dir/reldef.cpp.o' failed
gmake[2]: *** [storage/connect/CMakeFiles/connect.dir/reldef.cpp.o] Error 1
gmake[2]: Leaving directory '/usr/local/src/mariadb-10.1-aix-github/mariadb-aix-build'
CMakeFiles/Makefile2:809: recipe for target 'storage/connect/CMakeFiles/connect.dir/all' failed
gmake[1]: *** [storage/connect/CMakeFiles/connect.dir/all] Error 2
gmake[1]: Leaving directory '/usr/local/src/mariadb-10.1-aix-github/mariadb-aix-build'
Makefile:149: recipe for target 'all' failed
gmake: *** [all] Error 2`